src/Sulu/Bundle/AdminBundle/Command/DownloadLanguageCommand.php

```diff
@@ -83,29 +84,34 @@ protected function executeLanguage(InputInterface $input, OutputInterface $outpu
             return;
         }
 
-        $composerJson = \json_decode(\file_get_contents($this->projectDir . \DIRECTORY_SEPARATOR . 'composer.json'), true);
-        $packages = \array_keys($composerJson['require']);
-        $suluPackages = \array_filter($packages, function($package) {
-            return 0 === \strpos($package, 'sulu/');
-        });
+        /** @var array{require: array<string, string>} $composerJson */
+        $composerJson = \json_decode(
+            \file_get_contents($this->projectDir . \DIRECTORY_SEPARATOR . 'composer.json'),
+            true,
+            flags: JSON_THROW_ON_ERROR
+        );
 
         $translationsFile = $this->projectDir
             . \DIRECTORY_SEPARATOR
             . 'translations/sulu'
             . \DIRECTORY_SEPARATOR
             . 'admin.' . $language . '.json';
 
+        $translations = [];
         if (\file_exists($translationsFile)) {
             $translations = \json_decode(\file_get_contents($translationsFile), true);
-        } else {
-            $translations = [];
+            Assert::isArray($translations, 'Unable to read translations file into array: ' . $translationsFile);
         }
 
         $packages = [];
-        foreach ($suluPackages as $suluPackage) {
-            $packageTranslations = $this->downloadLanguage($output, $language, $suluPackage, $translationBaseUrl);
-            $packages[$suluPackage] = [
-                'Package' => $suluPackage,
+        foreach ($composerJson['require'] as $packageName => $version) {
+            if (!str_starts_with($packageName, 'sulu/')) {
+                continue;
+            }
+
+            $packageTranslations = $this->downloadLanguage($output, $language, $packageName, $translationBaseUrl);
+            $packages[$packageName] = [
+                'Package' => $packageName,
                 'Translations' => \count($packageTranslations),
             ];
             $translations = \array_merge($translations, $packageTranslations);
```

Comment:
Reading the composer file and iterating only once over it, seems like the smarter move. And this also crashes better if the composer file contains invalid json.

Reply:
It might even make sense to extract composer handling into it's own class for better typing. (This is not the only class that needs it.

Reply:
Reverted.